cli/reactjs_jsx_ppx_v3.ml

```diff
@@ -353,7 +353,14 @@ let jsxMapper () =
       | Ldot (_modulePath, value) as fullPath when isCap value -> Ldot (fullPath, "make")
       | modulePath -> modulePath
     in
-    let props = record in
+    let isEmptyRecord { pexp_desc } =
+      match pexp_desc with
+      | Pexp_record (labelDecls, _) when List.length labelDecls = 0 -> true
+      | _ -> false
+    in
+    (* check if record which goes to Foo.make({ ... } as record) empty or not
```

Comment:
What's an example of where this gets used?

Reply:
I should have written the comment in detail. 😄 
```rescript
@react.component
let make = () => <V4 />
```
⬇️ 
```rescript
@obj
type make = {key: option<string>}
let make = (_: make) => V4.make({})
//                              ^^^ build error
```
In this case, `unit` is also not working. So, I think the `_` seems proper as an argument.

Reply:
> I should have written the comment in detail. 😄
> 
> ```rescript
> @react.component
> let make = () => <V4 />
> ```
> 
> ⬇️
> 
> ```rescript
> @obj
> type make = {key: option<string>}
> let make = (_: make) => V4.make({})
> //                              ^^^ build error
> ```
> 
> In this case, `unit` is also not working. So, I think the `_` seems proper as an argument.

What does the generated code for the call look like now?
```foo(_)``` cannot be used I think, as it's not a valid id.

Reply:
This is the generated code now. What would be valid instead of `_`?
```rescript
module V4 = {
  @obj
  type make = {key: option<string>}

  let make = (_: make) => {
    ReactDOMRe.createElement(
      ReasonReact.fragment,
      [
        ReactDOMRe.createDOMElementVariadic(
          "div",
          ~props=ReactDOMRe.domProps(~id="1", ~onClick={_ => ()}, ()),
          [{`div`->React.string}],
        ),
      ],
    )
  }
}

@obj
type make = {key: option<string>}
let make = (_: make) => V4.make(_)
```

Reply:
Not sure. This is not about JSX per se.
If you were writing it all by hand, what would you use?
It seems there might be a gap in the language at the moment.
The best one can do, I guess it to pass `{key:None}`.

Reply:
Agree. I think `{key: None}` would be better than `_`. It's more generally acceptable. I'll fix it.

Reply:
Fixed it.
```rescript
@react.component
let make = () => <V4 />
```
⬇️ 
```rescript
@obj                   
type make = {
  key: option<string>,
}
let make = (_: make) => V4.make({key: None})
```
---
```rescript
@react.component
let make = () => <V4 key="key-v4" />
```
⬇️ 
```rescript
@obj                   
type make = {
  key: option<string>,
}
let make = (_: make) => V4.make({key: Some("key-v4")})
```